### DISCUSSION TITLE
Why node v21 engine is not supported in recent releases in v2.1.6 and v2.1.7?

### DISCUSSION THREAD

**Discussion by KumarVijayendra** [vitest-dev/vitest#7005] [General]
Saw this error while upgrading dependencies:

```console
error vitest@2.1.7: The engine "node" is incompatible with this module. Expected version "^18.0.0 || ^20.0.0 || >=22.0.0". Got "21.7.0"
error Found incompatible module.
```

Should this have been reverted in 2.1.7, which reverted 2.1.6 support for vite 6.

Snapshot of error: 
<img width="962" alt="vitest - node engine dependency error" src="https://github.com/user-attachments/assets/43577d44-2de2-48a4-84eb-3ee6fc8613e4">

**Comment by sheremet-va** [comment]
Node 21 reached end of life on 2024-06-01. The support was removed because Vite 6 doesn't support it anymore. After the revert, we should've also reverted the `engine` field. Should be fixed soon.

**Reply by KumarVijayendra** [reply]
Got it. Thanks for quick response. @sheremet-va

**Reply by Aashwin-Kumar** [reply]
the issue with the engines field in Vite hasn’t been fixed yet. Node 21 is still unsupported, and the install/runtime errors persist. Waiting on an official patch.
